### PR TITLE
Http evader v10

### PR DESCRIPTION
**Pull request by catenacyber**
Modifies #221 :
- Rebased and Removed already merged commits

### CODE REVIEW

**Comment by victorjulien** [comment]
I merged a number of 'safer looking' patches in #225

File: htp/htp_request_generic.c [hunk 1]
```diff
@@ -93,8 +93,8 @@ htp_status_t htp_process_request_header_generic(htp_connp_t *connp, unsigned cha
             // Don't use string comparison here because we want to
             // ignore small formatting differences.
 
-            int64_t existing_cl = htp_parse_content_length(h_existing->value);
-            int64_t new_cl = htp_parse_content_length(h->value);
+            int64_t existing_cl = htp_parse_content_length(h_existing->value, NULL);
```
Comment (victorjulien) [inline htp/htp_request_generic.c:96]: this commit looks like it does something else than the commit message says

Comment (catenacyber) [reply htp/htp_request_generic.c:96]: Indeed, with the latest merges, this one should now be squashed with 
282deb0

**Comment by catenacyber** [comment]
Great for #225

**Pull request closed without merging**